oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/contexts.js

```diff
@@ -1,8 +1,47 @@
-import React from 'react'
+import React, { useMemo, useContext } from "react";
+import { useFormikContext } from "formik";
+import { useFormConfig } from "./hooks";
+import { ApiClient, DepositActions } from "../api";
+console.log(useFormConfig);
 
-export const FormConfigContext = React.createContext()
+export const FormConfigContext = React.createContext();
 
+export const FormConfigProvider = ({ children, value }) => {
+  return (
+    <FormConfigContext.Provider value={value}>
+      {children}
+    </FormConfigContext.Provider>
+  );
+};
 
-export const FormConfigProvider = ({children, value}) => {
-    return <FormConfigContext.Provider value={value}>{children}</FormConfigContext.Provider>
-}
+export const ApiClientContext = React.createContext();
+
+export const ApiClientProvider = ({
+  children,
+  clientClass = DepositActions,
+}) => {
+  const {
+    formConfig: { createUrl },
+  } = useFormConfig();
+  // DefaultClass muze byt to OARepo.cosi..Client co ted mame
+  const formik = useFormikContext();
```

Comment:
same point as formConfig, here we kinda restrict any api client access to formik based children only (if this raises error elsewhere outside formik context). Would be nice if this could be moved to clientClass implementation where its actually needed and specific for current usage (form deposit actions). This also causes the following useMemo to recreate itself on each formik change

Reply:
@mirekys I fully agree with you, but I don't know how to make this happen. In current implementation, we specifically placed ApiClientProvider in DepositForm in location below BaseForm, so that I could call useFormikContext inside and pass it to the DepositActions (apiClient) class. DepositActions class, in its current implementation is 100% reliant on formik and cannot do much without it. Since I don't  see a way to gain access to formik directly from DepositActions class, I can either pass it here as it is now, or I could technically pass it to the action (individual method of depositAction class that I would be calling from buttons) in the button component (I could call useFormikContext there and then pass formik down). Do you have some suggestions please?